patches/server/1049-Add-paper-version-util-class.patch

```diff
@@ -0,0 +1,119 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: masmc05 <masmc05@gmail.com>
+Date: Fri, 16 Feb 2024 14:13:29 +0200
+Subject: [PATCH] Add paper version util class
+
+
+diff --git a/src/main/java/io/papermc/paper/util/misc/ServerInfoImpl.java b/src/main/java/io/papermc/paper/util/misc/ServerInfoImpl.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..dbd63f65c69f6ba97f4b2e264292d1229e2bb31a
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/misc/ServerInfoImpl.java
+@@ -0,0 +1,100 @@
++package io.papermc.paper.util.misc;
++
++import com.google.common.base.Preconditions;
++import io.papermc.paper.util.ServerInfo;
++import net.kyori.adventure.key.Key;
++import net.kyori.adventure.key.KeyPattern;
++import net.minecraft.SharedConstants;
++import org.bukkit.craftbukkit.util.Versioning;
++import org.jetbrains.annotations.NotNull;
++import java.util.HashSet;
++import java.util.Locale;
++import java.util.Set;
++import java.util.regex.Pattern;
++import java.util.stream.Stream;
++
++public class ServerInfoImpl implements ServerInfo {
++    private final String bukkitVersion = Versioning.getBukkitVersion();
++    private final String version = SharedConstants.getCurrentVersion().getId();
++    private final int[] versionArr = this.parseReleaseVersion(this.version);
++    private final Pattern release = Pattern.compile("[0-9](\\.[0-9]+)+");
```

Comment:
Is it a good idea to leave the length of this open ended? the `+` means that `1.2.3.4.5.6.7.8.9` is a valid version here. I think it should be more strict.

Reply:
It still could have worked and would have been parsed and correctly checked, but made it more strict